dartagnan/src/main/java/com/dat3m/dartagnan/program/event/core/Init.java

```diff
@@ -1,40 +1,69 @@
 package com.dat3m.dartagnan.program.event.core;
 
 import com.dat3m.dartagnan.expression.IConst;
-import com.dat3m.dartagnan.expression.IExpr;
 import com.dat3m.dartagnan.program.event.Tag;
 import com.dat3m.dartagnan.program.event.visitors.EventVisitor;
-
+import com.dat3m.dartagnan.program.memory.MemoryObject;
 import org.sosy_lab.java_smt.api.SolverContext;
 
+/**
+ * An instance of this class is generated for each defined location in the shared memory of a program.
+ * It is exposed to the memory consistency model as a visible event.
+ * It acts like a regular store, such that load events may read from it and other stores may overwrite it.
+ */
 public class Init extends MemEvent {
 
-	private final IConst value;
+	private final MemoryObject base;
+	private final int offset;
 	
-	public Init(IExpr address, IConst value) {
-		super(address, null);
-		this.value = value;
+	public Init(MemoryObject b, int o) {
+		super(b.add(o), null);
+		base = b;
+		offset = o;
 		addFilters(Tag.ANY, Tag.VISIBLE, Tag.MEMORY, Tag.WRITE, Tag.INIT);
 	}
 
+	/**
+	 * Partially identifies this instance in the program.
+	 * @return
+	 * Address of the array whose field is initialized in this event.
+	 */
+	public MemoryObject getBase() {
+		return base;
+	}
+
+	/**
+	 * Partially identifies this instance in the program.
+	 * @return
+	 * Number of fields before the initialized location.
```

Comment:
Before.  The initialized location may be the n-th field of an array, so there are n-1 fields located before it.  Its (relative) offset is the integer n-1.
There are usually multiple ways to explain classes and methods.  I always try to document the thing without directly using its name, because otherwise it would be just noise:
```
/**
 * Returns the offset.
 * @return
 * The offset of this.
 */
public int getOffset() {
```

Reply:
I think redundancy is not necessarily noise. Well chosen names for variables and methods etc. should be self-explanatory/very descriptive and this will necessarily make them overlap with their description (which is a good thing!). So your "offset" example is not bad, even if redundant (redundancy helps readers to understand).
The comment `Partially identifies this instance in the program.` on the other hand doesn't tell anything, especially if it is shared by other methods.

I would rather include the meaning of the class, e.g.
```
/**
 * Returns the memory object whose fields (see getOffset) gets initialized.
 * @return
 * The memory object to be initialized.
 */
public MemoryObject getBase() { // ... }

/**
 * Returns the offset of the field that gets initialized (relative to getBase)
 * @return
 * The field to be initialized
 */
public int getOffset() { // ... }


```